Detectors/TRD/calibration/src/T0Fit.cxx

```diff
@@ -0,0 +1,270 @@
+  }
+
+  mInitDone = true;
+}
+
+void T0Fit::retrievePrev(o2::framework::ProcessingContext& pc)
+{
+  // We either get a pointer to a valid object from the last ~hour or to the default object
+  // which is always present. The first has precedence over the latter.
+  auto dataCalT0 = pc.inputs().get<o2::trd::CalT0*>("calt0");
+  std::string msg = "Default Object";
+  // We check if the object we got is the default one by comparing it to the defaults.
+  for (int iDet = 0; iDet < constants::MAXCHAMBER; ++iDet) {
+    if (dataCalT0->getT0(iDet) != constants::T0DEFAULT) {
+      msg = "Previous Object";
+      break;
+    }
+  }
+  LOG(info) << "FitInstance: From CCDB retrieved " << msg;
+
+  // Here we set each entry regardless if it is the default or not.
+  for (int iDet = 0; iDet < constants::MAXCHAMBER; ++iDet) {
+    t0_chambers[iDet] = dataCalT0->getT0(iDet);
+  }
+  t0_average = dataCalT0->getT0av();
+}
+
+void T0Fit::finalizeSlot(Slot& slot)
+{
+  // do actual fits for the data provided in the given slot
+
+  TStopwatch timer;
+  timer.Start();
+  initProcessing();
+
+  // get data and fill profiles
+  auto dataPH = slot.getContainer();
+
+  for (int iEntry = 0; iEntry < dataPH->getNEntries(); ++iEntry) {
+    int iDet = dataPH->getDetector(iEntry);
+    adcProfIncl->Fill(dataPH->getTimeBin(iEntry), dataPH->getADC(iEntry));
+    adcProfDet[iDet]->Fill(dataPH->getTimeBin(iEntry), dataPH->getADC(iEntry));
+  }
+
+  // do fits
+  // inclusive distribution
+  mFitFunctor.x.clear();
+  mFitFunctor.y.clear();
+  for (int i = 1; i <= 30; ++i) {
+    mFitFunctor.x.push_back(i - 1);
+    mFitFunctor.y.push_back(adcProfIncl->GetBinContent(i));
+  }
+
+  mFitter.FitFCN();
+  auto fitResult = mFitter.Result();
+
+  if (fitResult.IsValid()) {
+    mFuncErfLandau->SetParameters(fitResult.GetParams()[0], fitResult.GetParams()[1], fitResult.GetParams()[2], fitResult.GetParams()[3]);
+    t0_average = mFuncErfLandau->GetMaximumX();
+  } else {
+    LOG(info) << "t0 fit for inclusive distribtion is not valid, set to -5";
+    t0_average = -5;
+  }
+
+  // single chambers
+  for (int iDet = 0; iDet < constants::MAXCHAMBER; ++iDet) {
+    if (adcProfDet[iDet]->GetEntries() < mMinEntriesChamber) {
+      LOG(info) << "not enough entries in chamber " << iDet << " for t0 fit, set to -5";
+      t0_chambers[iDet] = -5;
+      continue;
+    }
+
+    mFitFunctor.x.clear();
+    mFitFunctor.y.clear();
+    for (int i = 1; i <= 30; ++i) {
+      mFitFunctor.x.push_back(i - 1);
+      mFitFunctor.y.push_back(adcProfDet[iDet]->GetBinContent(i));
+    }
+
+    mFitter.FitFCN();
+    fitResult = mFitter.Result();
+
+    if (fitResult.IsValid()) {
+      mFuncErfLandau->SetParameters(fitResult.GetParams()[0], fitResult.GetParams()[1], fitResult.GetParams()[2], fitResult.GetParams()[3]);
+      t0_chambers[iDet] = mFuncErfLandau->GetMaximumX();
+    } else {
+      LOG(info) << "t0 fit for chamber " << iDet << " is not valid, set to -5";
+      t0_chambers[iDet] = -5;
```

Comment:
Maybe you can add a member to your fit class for the dummy value `int mDummyT0 = -5` and use that everywhere?

Reply:
added the dummy value